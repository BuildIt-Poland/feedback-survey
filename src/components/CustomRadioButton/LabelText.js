import styled from 'styled-components';

import media from '../../styles/media';
import { colorWhite, colorBlue } from '../../styles/designTokens';

const LabelText = styled.label`
  width: 15rem;
  height: 4.8rem;
  line-height: 4.8rem;
  text-transform: lowercase;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  margin: 0;
  border-radius: 1rem;

  ${media.phone`
    width: 25%;
    max-width: 15rem;
  `};

  input:not(:checked) + & {
    color: ${colorBlue};
    border: ${props => (props.withIcon ? 'none' : `0.2rem ${colorBlue} solid`)};
  }

  input:not(:checked) + &:hover,
  input:checked + & {
    color: ${colorWhite};
    background-color: ${props => (props.withIcon ? 'none' : `${colorBlue}`)};
    border: ${props => (props.withIcon ? 'none' : `0.2rem ${colorBlue} solid`)};
  }
`;

export default LabelText;
