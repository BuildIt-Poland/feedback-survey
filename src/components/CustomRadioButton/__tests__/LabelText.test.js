import React from 'react';
import { create } from 'react-test-renderer';

import LabelText from '../LabelText';

describe('COMPONENT - CustomRadioButton LabelText', () => {
  it('renders correctly', () => {
    const component = create(<LabelText />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with icon', () => {
    const component = create(<LabelText withIcon={true} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
