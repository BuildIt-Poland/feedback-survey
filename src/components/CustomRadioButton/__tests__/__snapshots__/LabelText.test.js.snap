// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - CustomRadioButton LabelText renders correctly 1`] = `
<label
  className="sc-bdVaJa iutOQo"
/>
`;

exports[`COMPONENT - CustomRadioButton LabelText renders correctly with icon 1`] = `
<label
  className="sc-bdVaJa kmKPE"
/>
`;
